cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -162,7 +178,12 @@ void Run(const std::vector<string>& bag_filenames) {
         break;
       }
 
-      // TODO(damonkohler): Republish non-conflicting tf messages.
+      // TODO(damonkohler): Check if republished tf messages are in conflict
```

Comment:
A conflicting transform is one that is both in the recorded bag and will be computed by the system. The logic here would have to duplicate the [logic in Node](https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/cartographer_ros/node.cc#L138) to determine which transforms to filter out.

It's not so difficult to do, but the duplication concerns me as it could lead to issues that may be difficult to debug if the logic ever changes.

Perhaps a better approach is to allow filtering out arbitrary transforms on the command line. We have a small [node that does this already](https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/scripts/tf_remove_frames.py).